src/Microsoft.TestPlatform.CrossPlatEngine/Execution/TestRunCache.cs

```diff
@@ -35,6 +35,8 @@ internal class TestRunCache : ITestRunCache
         /// </summary>
         private long totalExecutedTests;
 
+        private Dictionary<string, TestResultMetric> totalExecutedTestsPerAdapter = new Dictionary<string, TestResultMetric>();
```

Comment:
Could we make this a Concurrent Dictionary instead?

Reply:
I just copied the approach other the total tests property was using. I will consider it.